### ISSUE TITLE
Error during draft up

### ISSUE THREAD

**Issue by linsun** [Azure/draft#89]
Configured the repo to use docker hub.   Saw image pushed over.

Hit this error:

--> Deploying to Kubernetes
    Release "bald-newt" does not exist. Installing it now.
Could not install release: rpc error: code = Unknown desc = error converting YAML to JSON: yaml: did not find expected alphabetic or numeric character

**Comment by ultimateboy** [comment]
Can you provide a few more details such as which type of application you were building?

Thanks for the bug report. We'll look into this!

**Comment by linsun** [comment]
sorry, it is the app used in the tutorial:  draft/examples/python

**Comment by bacongobbler** [comment]
So to confirm, can you provide a few more details?

- your output of `helm version`
- your output of `draft version`
- your kubernetes cluster version
- the steps you followed to deploy the python example. Were you using the "Getting Started" guide?

**Comment by bacongobbler** [comment]
That yaml error means that somewhere in the `chart/` directory, one of the yaml manifests is not proper YAML (usually indentation errors). Have you made any changes to the example, or perhaps do you have an editor open that automatically changes tabs to spaces or vice versa? Those are the first things that come off the top of my head.

**Comment by linsun** [comment]
→ helm version
Client: &version.Version{SemVer:"v2.4.1", GitCommit:"46d9ea82e2c925186e1fc620a8320ce1314cbb02", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.4.1", GitCommit:"46d9ea82e2c925186e1fc620a8320ce1314cbb02", GitTreeState:"clean"}

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ✖︎ [15:51:50]
→ /Users/linsun/Downloads/darwin-amd64/draft version  
Client: &version.Version{SemVer:"canary", GitCommit:"8520032ef117cbbd64eff6f7e513c7cf8cd323f9", GitTreeState:"clean"}
Server: &version.Version{SemVer:"canary", GitCommit:"ed93463b23f1ebb5ee6f670b6dbbb51df11aa46e", GitTreeState:"clean"}

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ✖︎ [15:52:08]
→ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1", GitCommit:"b0b7a323cc5a4a2019b2e9520c21c7830b7f708e", GitTreeState:"clean", BuildDate:"2017-04-03T20:44:38Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.6-4+abe34653415733", GitCommit:"abe346534157336e6bd5a70702756cff19d43a49", GitTreeState:"clean", BuildDate:"2017-05-18T16:52:50Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}

**Comment by linsun** [comment]
I didn't make any changes and was just following through the tutorial.   These are 2 of my yaml files:

→ cat chart/Chart.yaml 
apiVersion: v1
description: A Helm chart for Kubernetes
name: bald-newt
version: 0.1.0

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ✖︎ [15:52:41]
→ cat chart/values.yaml 
# Default values for python.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.
replicaCount: 2
image:
  registry: docker.io
  org: library
  name: python
  tag: onbuild
  pullPolicy: IfNotPresent
service:
  name: python
  type: ClusterIP
  externalPort: 80
  internalPort: 8080
resources:
  limits:
    cpu: 100m
    memory: 128Mi
  requests:
    cpu: 100m
    memory: 128Mi

**Comment by linsun** [comment]
I just repeated the steps (by ensuring the python example dir is clean) and used `draft create` to recreate all the yamls.  Still hit the same error.

I'm also relatively new to helm.   I tried to install the generated chart:

```# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ● [16:55:53]
→ helm install chart
Error: release womping-kitten failed: Ingress.extensions "womping-kitten-awesome-kangaroo" is invalid: spec.rules[0].host: Invalid value: "womping-kitten.": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)* (e.g. 'example.com')```

**Comment by linsun** [comment]
Could someone shed some insight on how to install the generated chart manually?  

```# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python/chart on git:master ● [18:01:09]
→ helm lint
==> Linting .
[ERROR] Chart.yaml: directory name (chart) and chart name (awesome-kangaroo) must be the same
[INFO] Chart.yaml: icon is recommended

Error: 1 chart(s) linted, 1 chart(s) failed

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python/chart on git:master ● [18:01:12]
→ cd ..

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ● [18:02:33]
→ mv chart awesome-kangaroo

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python/awesome-kangaroo on git:master ✖︎ [18:02:46]
→ helm lint                
==> Linting .
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, no failures

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python/awesome-kangaroo on git:master ✖︎ [18:05:14]
→ cd ..

# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ✖︎ [18:05:50]
→ helm package awesome-kangaroo --debug
[debug] Saved /Users/linsun/go/src/github.com/Azure/draft/examples/python/awesome-kangaroo-0.1.0.tgz to current directory

[debug] Saved /Users/linsun/go/src/github.com/Azure/draft/examples/python/awesome-kangaroo-0.1.0.tgz to /Users/linsun/.helm/repository/local


# linsun at linsun in ~/go/src/github.com/Azure/draft/examples/python on git:master ✖︎ [18:05:57]
→ helm install awesome-kangaroo-0.1.0.tgz 
Error: release cranky-fish failed: Ingress.extensions "cranky-fish-awesome-kangaroo" is invalid: spec.rules[0].host: Invalid value: "cranky-fish.": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)* (e.g. 'example.com')```

**Comment by linsun** [comment]
ok, i got the python example chart installed manually via calling helm directly:

```→  helm install --set basedomain=draft.example.com chart-0.1.0.tgz 
NAME:   wanton-elephant
LAST DEPLOYED: Thu Jun  1 18:36:29 2017
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/Deployment
NAME                   DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
wanton-elephant-chart  2        2        2           0          3s

==> v1beta1/Ingress
NAME                   HOSTS                              ADDRESS      PORTS  AGE
wanton-elephant-chart  wanton-elephant.draft.example.com  50.23.5.244  80     3s

==> v1/Service
NAME                   CLUSTER-IP    EXTERNAL-IP  PORT(S)  AGE
wanton-elephant-chart  10.10.10.154  <none>       80/TCP   3s


NOTES:

  http://wanton-elephant.draft.example.com to access your application
```

**Comment by sgoings** [comment]
Woo! Yeah! Makin' progress!

What I don't quite understand is that you're able to install the chart with helm but not with draft which is really just shipping the files over to Tiller (server portion of Helm) and having Tiller do the same work that you just saw done with `wanton-elephant`.

Shot in the dark - but are you still having problems with `draft up` now that you're seeing success with a `helm install` of the same chart?

**Comment by linsun** [comment]
Still have probs.... in fact, tonight, I tried draft with mini kube and failed with the same error.

Could you provide what version of kube, helm, draft that works for you?  I'm on a MacOS.  

I've updated my helm to latest version.  See my versions:

```bash-3.2$ helm version
Client: &version.Version{SemVer:"v2.4.2", GitCommit:"82d8e9498d96535cc6787a6a9194a76161d29b4c", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.4.2", GitCommit:"82d8e9498d96535cc6787a6a9194a76161d29b4c", GitTreeState:"clean"}
bash-3.2$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1", GitCommit:"b0b7a323cc5a4a2019b2e9520c21c7830b7f708e", GitTreeState:"clean", BuildDate:"2017-04-03T20:44:38Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.0", GitCommit:"fff5156092b56e6bd60fff75aad4dc9de6b6ef37", GitTreeState:"dirty", BuildDate:"2017-04-07T20:46:46Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}
bash-3.2$ ~/Downloads/darwin-amd64/draft version
Client: &version.Version{SemVer:"canary", GitCommit:"8520032ef117cbbd64eff6f7e513c7cf8cd323f9", GitTreeState:"clean"}
Server: &version.Version{SemVer:"canary", GitCommit:"3fad7ef57ce9ed198649bef38023be3860792d24", GitTreeState:"clean"}```

**Comment by linsun** [comment]
one thing may be helpful to debugging, is this expected?  

bash-3.2$ cd examples/python/
bash-3.2$ ls
Dockerfile		app.py			chart			draft.toml		requirements.txt
bash-3.2$ cd chart/
bash-3.2$ helm lint
==> Linting .
[ERROR] Chart.yaml: directory name (chart) and chart name (billowing-cow) must be the same
[INFO] Chart.yaml: icon is recommended

Error: 1 chart(s) linted, 1 chart(s) failed

When I deploy helm chart, I had to get this error fixed (plus ensure it is the right registry, image name, tag, etc) before I can deploy.

**Comment by linsun** [comment]
Oh, one more thing.  I tried to update the chart.yaml file, and noticed the draft up console has connection refused error:

E0602 23:28:32.695890   15285 portforward.go:332] an error occurred forwarding 51188 -> 44135: error forwarding port 44135 to pod 5ef95025a31fc67f869192beeb93ac089d6096cf64324b0b6e0d5d31da98ce63, uid : exit status 1: 2017/06/03 03:28:32 socat[26457.140211455952640] E connect(5, AF=2 127.0.0.1:44135, 16): Connection refused
Error: there was an error running 'draft up': there was an error while dialing the server: unexpected EOF

**Comment by linsun** [comment]
I found the following in the tilt log (e.g. kubectl logs tiller-deploy-3354596499-kwxrz  --namespace=kube-system):

2017/06/05 14:54:40 storage.go:43: Getting release "bunking-butterfly" (v1) from storage
2017/06/05 14:54:40 storage.go:139: Getting release history for 'bunking-butterfly'
2017/06/05 14:54:40 release_server.go:660: Failed install prepare step: error converting YAML to JSON: yaml: did not find expected alphabetic or numeric character

Is there a way I can check out the release in the storage to see if it looks correct?  Given that I can use helm to deploy the chart, I think there may be an issue with uploading the chart to tiller via draft?

**Comment by linsun** [comment]
is there a online chat forum or slack that i can discuss this issue?

**Comment by bacongobbler** [comment]
Hi @linsun! Sorry for not getting back to you sooner. I have been away for 2 weeks on vacation while the announcement was going on. I am just getting back on my own two feet (and reading hundreds of emails/github notifications), but I will definitely take a look at this once I am back on my own two feet again. Sorry about the delayed response! I promise I haven't forgotten you. :)

**Comment by linsun** [comment]
thank you @bacongobbler!  would appreciate any insight from you on this.  I hit this pretty consistent on two kubernetes envs and 2 clients envs.

**Comment by bacongobbler** [comment]
So after debugging with @linsun over Slack, we found that it was due to a misconfigured base domain. When @linsun installed draft, she used `*.example.com` as the basedomain instead of `example.com`. This caused the helm chart to freak out due to the misconfigured ingress rule and raise the error seen above.

So, our action items for this ticket is to

1) improve the `draft init` experience, tracked in #49
2) document what exactly should be under the `basedomain` field as a crutch for now.
3) potentially look into how we can better improve helm template errors to show *where* the helm chart is failing to work. Being shown that the ingress template was failing to be parsed would have made debugging significantly easier.
